### PR TITLE
Gamerules

### PR DESCRIPTION
**Pull request by inxomnyaa**
Implemented and working gamerules:
- keepInventory
- doTileDrops
- doDaylightCycle
- showDeathMessages
  This is just the beginning.
  Plugins can create and access new gamerules and their values, or modify vanilla gamerules.
  API:
- Level::setGameRule($name, $value) -> returns if created
- Level::getGameRule($name) -> returns gamerule as string
  Also, gamerules are saved in level.dat, that means gamerules from imported worlds will work just perfect, and vise-versa when you export a world to a PC server.
  Changed gamerules save instantly into level.dat
  The command works like in Vanilla MCPC. /gamerule lists all gamerules.. but minimalistically differently. It also shows the currently values. /gamerule %rulename% shows the current value. /gamerule %rulename% %value% sets the value.
  Any questions? Feel free to ask.

### CODE REVIEW

**Comment by robske110** [comment]
See code comments

File: src/pocketmine/level/GameRules.php [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+		$this->addGameRule("doMobSpawning", true, ByteTag::class); // TODO: Add to AI branch
+		$this->addGameRule("doMobLoot", true, ByteTag::class); // TODO
+		$this->addGameRule("doTileDrops", true, ByteTag::class); // Implemented
+		$this->addGameRule("doEntityDrops", true, ByteTag::class); // TODO
+		$this->addGameRule("commandBlockOutput", true, ByteTag::class); // Not MCPE
+		$this->addGameRule("naturalRegeneration", true, ByteTag::class); // TODO
+		$this->addGameRule("doDaylightCycle", true, ByteTag::class); // Implemented
+		$this->addGameRule("logAdminCommands", true, ByteTag::class); // TODO
+		$this->addGameRule("showDeathMessages", true, ByteTag::class); // Implemented
+		$this->addGameRule("randomTickSpeed", 3, IntTag::class); // TODO
```
Comment (robske110) [inline src/pocketmine/level/GameRules.php:29]: I have found a way to do this, used it for my Grass development

File: src/pocketmine/level/GameRules.php [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+		$this->addGameRule("sendCommandFeedback", true, ByteTag::class); // TODO
+		$this->addGameRule("reducedDebugInfo", false, ByteTag::class); // TODO
```
Comment (robske110) [inline src/pocketmine/level/GameRules.php:31]: overwrite debug lvl 1/2

File: src/pocketmine/level/GameRules.php [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+		$this->addGameRule("spectatorsGenerateChunks", true, ByteTag::class); // TODO
+		$this->addGameRule("spawnRadius", 10, IntTag::class); // TODO
+		$this->addGameRule("disableElytraMovementCheck", false, ByteTag::class); // Not MCPE
```
Comment (robske110) [inline src/pocketmine/level/GameRules.php:34]: why even add this

File: src/pocketmine/level/GameRules.php (hunk 1 above)
Comment (inxomnyaa) [inline src/pocketmine/level/GameRules.php:29]: Yea. you can implement it later.

File: src/pocketmine/level/GameRules.php (hunk 2 above)
Comment (inxomnyaa) [inline src/pocketmine/level/GameRules.php:31]: sorry what.. thats what i said in group, right?

File: src/pocketmine/level/GameRules.php (hunk 3 above)
Comment (inxomnyaa) [inline src/pocketmine/level/GameRules.php:34]: We discussed this in the group. @dktapps got the point.

**Pull request merged**